core/src/main/java/org/lflang/federated/generator/FederateInstance.java

```diff
@@ -220,79 +204,87 @@ public Instantiation getInstantiation() {
   /** Keep a unique list of enabled serializers */
   public HashSet<SupportedSerializers> enabledSerializers = new HashSet<>();
 
+  /** Cached result of analysis of which reactions to exclude from main. */
+  private Set<Reaction> excludeReactions = null;
+
+  /** An error reporter */
+  private final MessageReporter messageReporter;
+
   /**
-   * Return true if the specified EObject should be included in the code generated for this
-   * federate.
+   * Return {@code true} if federate instance inherits the given reactor declaration.
    *
-   * @param object An {@code EObject}
-   * @return True if this federate contains the EObject.
+   * @param declaration The reactor declaration to inherit.
```

Comment:
This comment suggests that the function causes inheritance to occur, but the function is a predicate. I don't understand what you mean by inheritance here. A reactor inherits something if that something is defined in a reactor that it extends. But this seems to have nothing to do with that.